pubsub/iterator.go

```diff
@@ -438,6 +439,14 @@ func (it *messageIterator) sendModAck(m map[string]bool, deadline time.Duration)
 				recordStat(it.ctx, ModAckTimeoutCount, 1)
 				return nil
 			default:
+				if err == nil {
+					return nil
+				}
+				// This addresses an error where `context deadline exceeded` errors
+				// not captured by the previous case do not cause fatal errors.
+				if strings.Contains(err.Error(), "context deadline exceeded") {
```

Comment:
Maybe `xerrors.Is(err, context.DeadlineExceeded)`?

Reply:
So the `xerrors` approach doesn't currently work due to the way the error is injected into the fake. If there's an RPC error from `ModifyAckDeadline`, then the error will show up as `rpc error: code = DeadlineExceeded desc = context deadline exceeded`, so checking for context deadline exceeded wouldn't match in this case. I believe the error people are seeing _isn't_ an RPC error, but this case let's us check both. I'm ok with removing the test though, so using `xerrors.Is` makes sense.

Reply:
You could also keep the test and define a private `ServerReactorOption` that accepts an `error`? If the API for `WithErrorInjection` is still flexible, it might be worth having that take an error vs info to construct a grpc status.

Reply:
Yeah, I considered that, but it's still the fake server returning the error. Because of that, the grpc library will wrap any error with `rpc error: ...`